ratpack-core/src/main/java/ratpack/core/http/client/internal/RequestActionSupport.java

```diff
@@ -608,19 +608,22 @@ private static URI absolutizeRedirect(URI requestUri, String redirectLocation) t
         return URI.create(requestUri.getScheme() + ":" + redirectLocation);
       } else {
 
-        String path = redirectLocationUri.getPath();
+        String path = redirectLocationUri.getRawPath();
         if (!path.startsWith("/")) { // absolute path
-          path = getParentPath(requestUri.getPath()) + path;
+          path = getParentPath(requestUri.getRawPath()) + path;
         }
-        return new URI(
-          requestUri.getScheme(),
-          requestUri.getUserInfo(),
+
+        // By default, the URI encodes the path given to it. We don't want that here. But, we can let the URI
+        // do most of the work and then just append the raw path and query afterwords
+        String rawQuery = redirectLocationUri.getRawQuery() != null
+          ? "?" + redirectLocationUri.getRawQuery() : "";
+        return new URI(new URI(requestUri.getScheme(),
+          requestUri.getRawUserInfo(),
```

Comment:
If we're going to change the userinfo behavior, we should have tests for it.  I'd be fine with leaving it unchanged as well.

Reply:
I reverted that. It's not really in the spirit of what is being addressed here.